File: src/main/java/com/somefrills/events/InputEvent.java
package com.somefrills.events;

import com.somefrills.misc.KeyAction;
import net.minecraft.client.input.KeyInput;

public class InputEvent extends Cancellable {
    public int key, modifiers;
    public KeyAction action;
    public KeyInput keyInput;

    public InputEvent(KeyInput input, KeyAction action) {
        this.setCancelled(false);
        this.key = input.key();
        this.modifiers = input.modifiers();
        this.action = action;
        this.keyInput = input;
    }
}

File: src/main/kotlin/com/somefrills/events/InputEvent.kt
package com.somefrills.events

import com.somefrills.misc.KeyAction
import net.minecraft.client.input.KeyInput

class InputEvent(input: KeyInput, @JvmField val action: KeyAction) : Cancellable() {
    @JvmField
    var key: Int = input.key()
    var modifiers: Int = input.modifiers()
    var keyInput: KeyInput? = input
}
